azafea/event_processors/endless/metrics/events/__init__.py

```diff
@@ -791,6 +791,48 @@ def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
         return {'tables': payload.get_uint32()}
 
 
+class CacheHasInvalidElements(SingularEvent):
+    __tablename__ = 'cache_has_invalid_elements'
+    __event_uuid__ = 'cbfbcbdb-6af2-f1db-9e11-6cc25846e296'
+    __payload_type__ = '(tt)'
+
+    # These come in as uint64, but the values won’t reach the limit of a BIGINT (int64)
+    number_of_valid_elements = Column(BigInteger, nullable=False)
+    number_of_bytes_read = Column(BigInteger, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'number_of_valid_elements': payload.get_child_value(0).get_uint64(),
```

Comment:
What would happen if an event was submitted with a value greater than the upper bound of an int64? Presumably undesirable/unpredictable results.

Rather than just assuming it won't happen, let's define a function like:

```python3
def clamp_to_int64(u64):
    return min(u64, GLib.MAXINT64)
```

and then:

```suggestion
            'number_of_valid_elements': clamp_to_int64(payload.get_child_value(0).get_uint64()),
```

etc.

Reply:
> Rather than just assuming it won't happen, let's define a function like:

Of course! It’s just a draft PR for now, I don’t even know if it works with real data!